File: backend/MemoryGameApi/Models/Game.cs
namespace MemoryGameApi.Models
{
    public enum GameStatus
    {
        NotStarted,
        InProgress,
        Completed
    }

    public class Game
    {
        public string Id { get; set; } = Guid.NewGuid().ToString();
        public List<Card> Cards { get; set; } = new List<Card>();
        public GameStatus Status { get; set; } = GameStatus.NotStarted;
        public int Moves { get; set; }
        public int MatchedPairs { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? CompletedAt { get; set; }
        public TimeSpan? Duration => CompletedAt?.Subtract(CreatedAt);
    }
}

File: frontend/memory-game/src/types/game.ts
export interface Card {
  id: number;
  value: string;
  isFlipped: boolean;
  isMatched: boolean;
}

export enum GameStatus {
  NotStarted = 0,
  InProgress = 1,
  Completed = 2
}

export interface Game {
  id: string;
  cards: Card[];
  status: GameStatus;
  moves: number;
  matchedPairs: number;
  createdAt: string;
  completedAt?: string;
  duration?: string;
  durationFormatted?: string;
  isGameComplete: boolean;
}

export interface CreateGameRequest {
  gridSize: number;
}

export interface FlipCardRequest {
  cardId: number;
}